google/cloud/storage/async_connection.h

```diff
@@ -29,6 +29,7 @@ namespace google {
 namespace cloud {
 namespace storage_experimental {
 GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+class AsyncReaderConnection;
```

Comment:
Why choose to do that over including "google/cloud/storage/async_reader_connection.h", and despite it only being used by name?

Reply:
No particular reason to chose one over the other (I mean, marginally faster compilation in some TUs, but I doubt we could measure that).

Reply:
Coffee talk: Perhaps we should prefer not to use forward declarations in general unless they're required or there is some tangible benefit (go/totw/192 internally).